src/ovirtapi/transform.js

```diff
@@ -51,6 +51,10 @@ function convertBool (val: ?string): boolean {
   return val ? val.toLowerCase() === 'true' : false
 }
 
+function convertToBoolString (val: ?boolean): BoolStringType {
```

Comment:
For converting truthy values it's better to use `Boolean(val).toString()` . No need for custom function here.

Reply:
So if we decide to go this way, we need to change that for all other API Boolean attributes handled on web-ui (which are declared now as Boolean instead of Strings). 
Otherwise the code is not consistent. Check: https://github.com/oVirt/ovirt-web-ui/pull/1386/#discussion_r611568797.
I can add a separate issue for that.

Reply:
Sending JSON booleans works just fine for anything value that is a "Boolean".  But...the engine will send back the field as a string that we transform back to a boolean.

For this field specifically, here is the JAXB generated Java code for the rest api model object:
```java
    @XmlElement(name = "persist_memorystate", type = String.class)
    @XmlJavaTypeAdapter(Adapter1 .class)
    @XmlSchemaType(name = "boolean")
    protected Boolean persistMemorystate;
```

The `Adapter1` class, also generated by JAXB, handles mapping Strings to Boolean and Boolean back to String:
```java
public class Adapter1
    extends XmlAdapter<String, Boolean>
{
    public Boolean unmarshal(String value) {
        return (org.ovirt.engine.api.utils.BooleanParser.parseBoolean(value));
    }

    public String marshal(Boolean value) {
        if (value == null) {
            return null;
        }
        return (javax.xml.bind.DatatypeConverter.printBoolean((boolean)(boolean)value));
    }
}
```

And the `BooleanParser` looks like this:
```java
public class BooleanParser {
    public static boolean parseBoolean(String value) {
        if ("true".equalsIgnoreCase(value) || "1".equals(value)) {
            return true;
        }
        if ("false".equalsIgnoreCase(value) || "0".equals(value)) {
            return false;
        }
        throw new InvalidValueException(
            "Value \"" + value + "\" isn't a valid boolean, it should be \"true\" , \"false\", \"0\" or \"1\""
        );
    }
}
```

The mismatch is how we define the value's type on our Api flow type.  Since it'll be received as "true" or "false" from the engine, it is technically a string.  Probably the best course of action is to make sure that all boolean types on the Api side should be (1) a generic string, or (2) a `string | boolean`, or (3) a `ApiBooleanType = "true" | "false"`.  Following that, all `toApi` functions would then use a function like `toApiBoolean(value)` so all boolean are handled the same way.

```js
function toApiBoolean (value: any): ApiBooleanType {
  return Boolean(value).toString()
}
```

(sorry for being long winded)

Reply:
> The mismatch is how we define the value's type on our Api flow type. Since it'll be received as "true" or "false" from the engine, it is technically a string. Probably the best course of action is to make sure that all boolean types on the Api side should be (1) a generic string, or (2) a string | boolean, or (3) a ApiBooleanType = "true" | "false". 

We can treat the value's type on our API flow as a Boolean since we know it should work as expected or as a ApiBooleanType = "true" | "false" just to avoid confusion for next time.

> Following that, all toApi functions would then use a function like toApiBoolean(value) so all boolean are handled the same way.
> 
> function toApiBoolean (value: any): ApiBooleanType {
>   return Boolean(value).toString()
> }

If we'll choose to define as ApiBooleanType then we should fix all place in a followup PR.